### ISSUE TITLE
KeyError when trying to compute/persist multiple keys

### ISSUE THREAD

**Issue by cvb** [dask/dask#4458]
Hi, having this code:
```
import dask
from dask.bag import from_sequence, zip as bag_zip

a = from_sequence(range(10), npartitions=2)
b = from_sequence(range(10), npartitions=2)
c = bag_zip(a, b)
d1 = c.pluck(0)
d2 = c.pluck(1)
e = d1.map(lambda x: x+1)

dask.persist(d1,d2,e)
```
I'm getting this error:
```
Traceback (most recent call last):
  File "key_error.py", line 11, in <module>
    dask.compute(d1,d2,e)
  File "/home/cvb/src/dask/dask/base.py", line 398, in compute
    results = schedule(dsk, keys, **kwargs)
  File "/home/cvb/src/dask/dask/multiprocessing.py", line 173, in get
    dsk2, dependencies = cull(dsk, keys)
  File "/home/cvb/src/dask/dask/optimization.py", line 46, in cull
    for k in work]
  File "/home/cvb/src/dask/dask/optimization.py", line 46, in <listcomp>
    for k in work]
  File "/home/cvb/src/dask/dask/core.py", line 187, in get_dependencies
    arg = dsk[key]
KeyError: ('pluck-ddca17d615f33c398bce4e491736fb48', 0)
```
Can reproduce this with 1.1.0 version and with current master branch. Having same result when using `dask.compute` instead of `dask.persist`.

**Comment by mrocklin** [comment]
Thanks for the excellent bug report and reproducible example @cvb .  I'm able to reproduce it.

@eriknw any time and interest in looking into this?

**Comment by jcrist** [comment]
Thanks for the bug report. This is fixed in #4464.

```python
In [1]: %load test.py

In [2]: # %load test.py
   ...: import dask
   ...: import dask.bag as db
   ...:
   ...: def inc(x):
   ...:     return x + 1
   ...:
   ...: a = db.from_sequence(range(10), npartitions=2)
   ...: b = db.from_sequence(range(10), npartitions=2)
   ...: c = db.zip(a, b)
   ...: d1 = c.pluck(0)
   ...: d2 = c.pluck(1)
   ...: e = d1.map(inc)
   ...:
   ...: out = dask.compute(d1, d2, e)
   ...:

In [3]: out
Out[3]:
([0, 1, 2, 3, 4, 5, 6, 7, 8, 9],
 [0, 1, 2, 3, 4, 5, 6, 7, 8, 9],
 [1, 2, 3, 4, 5, 6, 7, 8, 9, 10])
```

**Comment by cvb** [comment]
Awesome, thanks!
